src/component-library/Input/BaseInput.tsx

```diff
@@ -0,0 +1,75 @@
+import { forwardRef, InputHTMLAttributes, ReactNode } from 'react';
+
+import { HelperText, HelperTextProps } from '../HelperText';
+import { hasErrorMessage } from '../HelperText/HelperText';
+import { Label, LabelProps } from '../Label';
+import { Adornment, BaseInputWrapper, StyledBaseInput, Wrapper } from './Input.style';
+
+type Props = {
+  label?: ReactNode;
+  labelProps?: LabelProps;
+  startAdornment?: ReactNode;
+  endAdornment?: ReactNode;
+  value?: string | ReadonlyArray<string> | number;
+  defaultValue?: string | ReadonlyArray<string> | number;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+type NativeAttrs = Omit<InputHTMLAttributes<HTMLInputElement>, keyof Props>;
+
+type InhriteAttrs = Omit<HelperTextProps, keyof Props & NativeAttrs>;
+
+type BaseInputProps = Props & NativeAttrs & InhriteAttrs;
+
+const BaseInput = forwardRef<HTMLInputElement, BaseInputProps>(
+  (
+    {
+      className,
+      style,
+      hidden,
+      startAdornment,
+      endAdornment,
+      label,
+      labelProps,
+      errorMessage,
+      errorMessageProps,
+      description,
+      descriptionProps,
+      disabled,
+      ...props
+    },
+    ref
+  ): JSX.Element => {
+    const hasError = hasErrorMessage(errorMessage);
+    const hasHelpText = !!description || hasError;
+
+    return (
+      <Wrapper hidden={hidden} className={className} style={style}>
+        {label && <Label {...labelProps}>{label}</Label>}
+        <BaseInputWrapper
+          $hasStartAdornment={!!startAdornment}
+          $hasEndAdornment={!!endAdornment}
+          $hasError={hasError}
+          $isDisabled={disabled}
+        >
+          {startAdornment && <Adornment>{startAdornment}</Adornment>}
+          <StyledBaseInput $isDisabled={disabled} disabled={disabled} ref={ref} type='text' {...props} />
+          {endAdornment && <Adornment>{endAdornment}</Adornment>}
+        </BaseInputWrapper>
+        {hasHelpText && (
+          <HelperText
```

Comment:
Is there a situation where we would want to show the helper text and error text at the same time? Not entirely sure about the UX of replacing descriptive text with an error when validating (although ultimately this is a decision for Jay)

Reply:
Normally you dont have both at the same time, but we can check with him later.

Reply:
Fine then :) this works as a solution for implementing the UX as designed. We can always refactor if we need to.